pkg/gadgets/internal/socketenricher/bpf/socket-enricher.bpf.c

```diff
@@ -25,17 +26,7 @@ struct {
 	__type(value, void *);
 } start SEC(".maps");
 
-const volatile __u64 socket_file_ops_addr = 0;
-
-static __always_inline void
-prepare_socket_key(struct sockets_key *socket_key, struct sock *sock)
-{
-	struct inet_sock *inet_sock = (struct inet_sock *)sock;
-	BPF_CORE_READ_INTO(&socket_key->netns, sock, __sk_common.skc_net.net, ns.inum);
-	BPF_CORE_READ_INTO(&socket_key->family, sock, __sk_common.skc_family);
-	socket_key->proto = BPF_CORE_READ_BITFIELD_PROBED(sock, sk_protocol);
-	socket_key->port = bpf_ntohs(BPF_CORE_READ(inet_sock, inet_sport));
-}
+const volatile __u64 disable_bpf_iterators = 0;
```

Comment:
Can it be a boolean?

Reply:
I am not sure: `bool` is not a built-in type, and userspace programs are supposed to `#include <stdbool.h>` to make use of it. In eBPF, we have vmlinux.h defining `bool`:
```
typedef _Bool bool;
```
`_Bool` is defined in C99.

I have not tested if cilium/ebpf's `RewriteConstants()` works with this type.
I see cilium/ebpf's `marshalBytes()` uses `binary.Write()`, which according to its documentation:
> Boolean values encode as one byte: 1 for true, and 0 for false.

Should I try it?

Reply:
It should work fine. Other gadgets are using bool too:

```bash
$ git grep "const volatile bool"
pkg/gadgets/common/mntns_filter.h:const volatile bool gadget_filter_by_mntns = false;
pkg/gadgets/profile/block-io/tracer/bpf/biolatency.bpf.c:const volatile bool filter_cg = false;
pkg/gadgets/profile/block-io/tracer/bpf/biolatency.bpf.c:const volatile bool targ_per_disk = false;
pkg/gadgets/profile/block-io/tracer/bpf/biolatency.bpf.c:const volatile bool targ_per_flag = false;
pkg/gadgets/profile/block-io/tracer/bpf/biolatency.bpf.c:const volatile bool targ_queued = false;
pkg/gadgets/profile/block-io/tracer/bpf/biolatency.bpf.c:const volatile bool targ_ms = false;
pkg/gadgets/profile/block-io/tracer/bpf/biolatency.bpf.c:const volatile bool filter_dev = false;
pkg/gadgets/profile/cpu/tracer/bpf/profile.bpf.c:const volatile bool kernel_stacks_only = false;
pkg/gadgets/profile/cpu/tracer/bpf/profile.bpf.c:const volatile bool user_stacks_only = false;
pkg/gadgets/profile/cpu/tracer/bpf/profile.bpf.c:const volatile bool include_idle = false;
pkg/gadgets/profile/tcprtt/tracer/bpf/tcprtt.bpf.c:const volatile bool targ_laddr_hist = false;
pkg/gadgets/profile/tcprtt/tracer/bpf/tcprtt.bpf.c:const volatile bool targ_raddr_hist = false;
pkg/gadgets/profile/tcprtt/tracer/bpf/tcprtt.bpf.c:const volatile bool targ_ms = false;
pkg/gadgets/snapshot/process/tracer/bpf/process-collector.bpf.c:const volatile bool show_threads = false;
pkg/gadgets/top/file/tracer/bpf/filetop.bpf.c:const volatile bool regular_file_only = true;
pkg/gadgets/trace/bind/tracer/bpf/bindsnoop.bpf.c:const volatile bool ignore_errors = true;
pkg/gadgets/trace/bind/tracer/bpf/bindsnoop.bpf.c:const volatile bool filter_by_port = false;
pkg/gadgets/trace/capabilities/tracer/bpf/capable.bpf.c:const volatile bool audit_only = false;
pkg/gadgets/trace/capabilities/tracer/bpf/capable.bpf.c:const volatile bool unique = false;
pkg/gadgets/trace/exec/tracer/bpf/execsnoop.bpf.c:const volatile bool ignore_failed = true;
pkg/gadgets/trace/open/tracer/bpf/opensnoop.bpf.c:const volatile bool targ_failed = false;
pkg/gadgets/trace/signal/tracer/bpf/sigsnoop.bpf.c:const volatile bool failed_only = false;
pkg/gadgets/trace/tcpconnect/tracer/bpf/tcpconnect.bpf.c:const volatile bool do_count = 0;
pkg/gadgets/trace/tcpconnect/tracer/bpf/tcpconnect.bpf.c:const volatile bool calculate_latency = false;
```